Biology.java

```diff
@@ -0,0 +1,346 @@
+        }
+    }
+
+    /**
+     * DNA and RNA codon table
+     */
+    public enum CodonMap {
+        CODONS();
+        public final Map<String, String> codonMap;
+
+        CodonMap() {
+            this.codonMap = new HashMap<>() {{
+                put("TTT", "F");
+                put("TTC", "F");
+                put("TTA", "L");
+                put("TTG", "L");
+
+                put("CTT", "L");
+                put("CTC", "L");
+                put("CTA", "L");
+                put("CTG", "L");
+
+                put("ATT", "I");
+                put("ATC", "I");
+                put("ATA", "I");
+                put("ATG", "M");
+
+                put("GTT", "V");
+                put("GTC", "V");
+                put("GTA", "V");
+                put("GTG", "V");
+
+
+                put("TCT", "S");
+                put("TCC", "S");
+                put("TCA", "S");
+                put("TCG", "S");
+
+                put("CCT", "P");
+                put("CCC", "P");
+                put("CCA", "P");
+                put("CCG", "P");
+
+                put("ACT", "T");
+                put("ACC", "T");
+                put("ACA", "T");
+                put("ACG", "T");
+
+                put("GCT", "A");
+                put("GCC", "A");
+                put("GCA", "A");
+                put("GCG", "A");
+
+
+                put("TAT", "Y");
+                put("TAC", "Y");
+                put("TAA", "STOP");
+                put("TAG", "STOP");
+
+                put("CAT", "H");
+                put("CAC", "H");
+                put("CAA", "Q");
+                put("CAG", "Q");
+
+                put("AAT", "N");
+                put("AAC", "N");
+                put("AAA", "K");
+                put("AAG", "K");
+
+                put("GAT", "D");
+                put("GAC", "D");
+                put("GAA", "E");
+                put("GAG", "E");
+
+
+                put("TGT", "C");
+                put("TGC", "C");
+                put("TGA", "STOP");
+                put("TGG", "W");
+
+                put("CGT", "R");
+                put("CGC", "R");
+                put("CGA", "R");
+                put("CGG", "R");
+
+                put("AGT", "S");
+                put("AGC", "S");
+                put("AGA", "T");
+                put("AGG", "T");
+
+                put("GGT", "G");
+                put("GGC", "G");
+                put("GGA", "G");
+                put("GGG", "G");
+            }};
+        }
+
+        /**
+         * Translates and returns the amino acid specified by the given codon
+         *
+         * @param codon the sequence of three nucleotides to translate
+         * @return the corresponding amino acid
+         */
+        public String getAminoAcid(String codon) {
+            return this.codonMap.get(codon);
+        }
+    }
+
+    /**
+     * Matrices of vectors to determine where to move virus in Euclidean space
```

Comment:
Can we be more descriptive here? Like: "The below code determines how changes in sequence lead to changes in to a virus's location in antigenic space. Specifically, for each site we pre-compute a vector for each possible amino-acid mutations. Vectors angles are drawn from a uniform distribution, while vector magnitudes are drawn from a gamma distribution. Gamma distributions differ for epitope and non-epitope sites.